src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -674,6 +675,76 @@ ss::future<init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
       });
 }
 
+namespace {
+
```

Comment:
nit: Can you preserve the commit message somewhere in the code? The semantics here are really intricate and difficult to comprehend without context (even better to add a link to KIP-360 proposal).